### PR TITLE
Update CommandDispatcher.java

### PR DESCRIPTION
**Pull request by tanmaymanolkar1**
builds upon
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1329

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java [hunk 1]
```diff
@@ -516,7 +517,10 @@ private static boolean eligibleToCache(CommandResult commandResult) {
     private static boolean eligibleToCacheException(BaseException exception) {
         //TODO : ADO 1373343 Add the whole transient exception category.
         if (exception instanceof IntuneAppProtectionPolicyRequiredException
-                || ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException)exception).getErrorCode())) {
+                || ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException) exception).getErrorCode())
+                || BrokerCommunicationException.Category.CONNECTION_ERROR.toString().equals(((BrokerCommunicationException) exception).getCategory())
+                || ClientException.INTERRUPTED_OPERATION.equals(((ClientException) exception).getErrorCode())
+                || ClientException.IO_ERROR.equals(((ClientException) exception).getErrorCode())) {
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:523]: Can we just put these in a `Set` and do a contains check on that?

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java [hunk 2]
```diff
@@ -516,7 +517,10 @@ private static boolean eligibleToCache(CommandResult commandResult) {
     private static boolean eligibleToCacheException(BaseException exception) {
         //TODO : ADO 1373343 Add the whole transient exception category.
         if (exception instanceof IntuneAppProtectionPolicyRequiredException
-                || ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException)exception).getErrorCode())) {
+                || ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException) exception).getErrorCode())
+                || BrokerCommunicationException.Category.CONNECTION_ERROR.toString().equals(((BrokerCommunicationException) exception).getCategory())
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:521]: If you've got to cast, here you're going to need instanceof

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java (hunk 1 above)
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:523]: Casting shouldn't be needed apart from the `BrokerCommunicationException`

Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:523]: Yeah, there's no need to cast these.

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java [hunk 3]
```diff
@@ -95,6 +98,14 @@
     private static InteractiveTokenCommand sCommand = null;
     private static final CommandResultCache sCommandResultCache = new CommandResultCache();
 
+
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:101]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java [hunk 4]
```diff
@@ -514,9 +525,15 @@ private static boolean eligibleToCache(CommandResult commandResult) {
      * @return
      */
     private static boolean eligibleToCacheException(BaseException exception) {
+        final String mErrorCode;
+        if (exception instanceof BrokerCommunicationException) {
+            mErrorCode = ((BrokerCommunicationException) exception).getCategory().toString();
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:530]: check category for null

Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:530]: nevermind, it isn't non null, but it's dereferenced in the contstructor.

**Review by AdamBJohnsonx — APPROVED** [review]
minor nitpick.

**Review by p3dr0rv — APPROVED** [review]

**Pull request merged**